app/lib/publisher/models.dart

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:gcloud/db.dart' as db;
+
+/// Canonical publisher data.
+@db.Kind(name: 'Publisher', idType: db.IdType.String)
+class Publisher extends db.ExpandoModel {
+  /// The associated domain name of the publisher.
+  String get domainName => id as String;
+
+  /// Markdown formatted description of the publisher.
+  ///
+  /// Limited to 64 KB in length.
+  @db.StringProperty()
+  String description;
+
+  @db.StringProperty()
+  String websiteUrl;
+
+  @db.DateTimeProperty()
+  DateTime created;
+
+  @db.DateTimeProperty()
+  DateTime updated;
+}
+
+/// Derived publisher data.
+@db.Kind(name: 'PublisherInfo', idType: db.IdType.String)
+class PublisherInfo extends db.ExpandoModel {
+  /// The associated domain name of the publisher.
+  String get domainName => id as String;
+
+  @db.DateTimeProperty()
+  DateTime updated;
+
+  /// List of packages that are associated with this publisher.
+  @db.StringListProperty()
+  List<String> packages;
+
+  /// List of userIds that are administrators of this publisher.
+  /// (their e-mail address is public information)
+  @db.StringListProperty()
+  List<String> admins;
+
+  /// List of userIds that are public members of this publisher.
+  /// (their e-mail address is public information)
+  @db.StringListProperty()
+  List<String> publicMembers;
+}
+
+/// Values for [PublisherMember.role]
+abstract class PublisherMemberRole {
+  /// Administrator of the publisher.
+  static const String admin = 'admin';
+
+  /// Accepted membership.
+  static const String member = 'member';
+
+  /// Invited but not accepted.
+  static const String pending = 'pending';
+}
+
+/// Stores the membership information of a single user.
+@db.Kind(name: 'PublisherMember', idType: db.IdType.String)
```

Comment:
Why don't we need a nonce? Are you assuming people have logged into to pub before they were invited to join a publisher?

Or do we just want to rely on the email verification that Google does in it's oauth?

Reply:
Yes, I assumed that once we have login, we could remove the nonce-based notifications: we can still send an e-mail, but we require a login, and then the user clicks the approve button.

Reply:
Fair point, we trust that Google accounts have verified emails.